Source/PhantomJs.NetCore/PdfGenerator.cs

```diff
@@ -0,0 +1,120 @@
+
+      try
+      {
+        var exeToUse = GetOsExecutableName();
+        WriteResourcesToDisk(exeToUse);
+        WriteResourcesToDisk(Consts.Rasterize);
+        pdfFileName = ExecutePhantomJs(exeToUse, htmlFileName, outputFolder, param);
+      }
+      finally
+      {
+        File.Delete(Path.Combine(PhantomRootFolder, htmlFileName));
+      }
+
+      return pdfFileName;
+    }
+
+    private string GetOsExecutableName() =>
+      RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? Consts.WinEXE
+      : RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? Consts.LinuxEXE
+      : Consts.OSXEXE;
+
+    private string WriteHtmlToTempFile(string html)
+    {
+      var filename = Path.GetRandomFileName() + ".html";
+      var absolutePath = Path.Combine(PhantomRootFolder, filename);
+
+      File.WriteAllText(absolutePath, html);
+
+      return filename;
+    }
+
+    private string ExecutePhantomJs(string phantomJsExeToUse, string inputFileName, string outputFolder, PdfGeneratorParams param)
+    {
+      if (param == null) param = new PdfGeneratorParams();
+
+      var layout = param.PageWidth > 0 && param.PageHeight > 0
+        ? $"{param.PageWidth}{param.DimensionUnit.GetValue()}*{param.PageHeight}{param.DimensionUnit.GetValue()}"
+        : param.Format.ToString();
+      var outputFileName = Path.GetFileNameWithoutExtension(inputFileName);
+      var outputFilePath = Path.Combine(outputFolder, $"{outputFileName}.pdf");
+      var exePath = Path.Combine(PhantomRootFolder, phantomJsExeToUse);
+
+      var startInfo = new ProcessStartInfo
+      {
+        FileName = exePath,
+        WorkingDirectory = PhantomRootFolder,
+        Arguments = $@"rasterize.js ""{inputFileName}"" ""{outputFilePath}"" ""{layout}"" {param.ZoomFactor}",
+          // TODO: include orientation parameter ""{param.Orientation.GetValue()}"" ",
+        UseShellExecute = false,
+        CreateNoWindow = true
+      };
+
+      var proc = new Process { StartInfo = startInfo };
+      proc.Start();
+      proc.WaitForExit();
+
+      return outputFilePath;
+    }
+
+    private void WriteResourcesToDisk(string exeFile)
+    {
+      if (File.Exists(exeFile)) return;
+
+      var assembly = Assembly.GetExecutingAssembly();
+      var zipExeFile = $"{Path.GetFileNameWithoutExtension(exeFile)}.zip";
+      var resourcePath = $"PhantomJs.NetCore.Resources.{zipExeFile}";
+      var outputPath = Path.Combine(PhantomRootFolder, zipExeFile);
+
+      using (var stream = assembly.GetManifestResourceStream(resourcePath))
+      using (var binaryReader = new BinaryReader(stream))
+      using (var fileStream = new FileStream(outputPath, FileMode.Create))
+      using (var binaryWriter = new BinaryWriter(fileStream))
+      {
+        byte[] byteArray = new byte[stream.Length];
+        stream.Read(byteArray, 0, byteArray.Length);
+        binaryWriter.Write(byteArray);
+      }
+
+      ZipFile.ExtractToDirectory(zipExeFile, PhantomRootFolder);
```

Comment:
This line does not work on my system (OSX with .net core 3.1), as it will look for the zip file in the folder where the .csproj file is located (I guess because I run `dotnet run` there).